sql/src/main/java/io/crate/expression/scalar/string/EncodeDecodeFunction.java

```diff
@@ -0,0 +1,126 @@
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import io.crate.common.Hex;
+import io.crate.common.Octal;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.expression.scalar.arithmetic.BinaryScalar;
+import io.crate.metadata.FunctionInfo;
+import io.crate.types.DataTypes;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Locale;
+import java.util.function.BinaryOperator;
+
+public class EncodeDecodeFunction {
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(new BinaryScalar<>(new Encode(), "encode", DataTypes.STRING, FunctionInfo.DETERMINISTIC_ONLY));
+        module.register(new BinaryScalar<>(new Decode(), "decode", DataTypes.STRING, FunctionInfo.DETERMINISTIC_ONLY));
+    }
+
+    /**
+     * Takes a binary data of "bytea" type and encodes it into the given output format.
+     */
+    private static class Encode implements BinaryOperator<String> {
+
+        @Override
+        public String apply(String bytea, String format) {
+            switch (Format.valueOf(format.toUpperCase(Locale.ROOT))) {
+                case BASE64:
+                    final byte[] text;
+                    if (Hex.isHexFormat(bytea)) {
+                        text = Hex.decodeHex(Hex.stripHexFormatFlag(bytea));
+                    } else {
+                        text = Octal.decode(bytea);
+                    }
+                    return Base64.getEncoder().encodeToString(text);
+                case HEX:
+                    if (Hex.isHexFormat(bytea)) {
+                        // the input is already in hex format
+                        return Hex.validateHex(Hex.stripHexFormatFlag(bytea));
+                    }
+                    return Hex.encodeHexString(Octal.decode(bytea));
+                case ESCAPE:
+                    if (Hex.isHexFormat(bytea)) {
+                        return Octal.encode(Hex.decodeHex(Hex.stripHexFormatFlag(bytea)));
+                    }
+                    return Octal.encode(bytea.getBytes(StandardCharsets.UTF_8));
+                default:
+                    // should not happen
+                    throw new RuntimeException();
```

Comment:
By defining a `encode` and `decode` function to the `Format` enum itself, this switch statement can be eliminated which improves performance.
Also the error handling should be improved, like e.g. catching the not-found enum.valueOf exception and turn it into a user friendly "Format 'foo' not supported" exception.